### ISSUE TITLE
False positive `no-useless-method` trigger

### ISSUE THREAD

**Issue by Kelin2025** [effector/eslint-plugin#74]
This code
```tsx
$nameValue.on(
  guard({
    clock: $name,
    filter: $shouldSyncRealName,
  }),
  (prev, newRealName) => newRealName
)
```
Results in
![image](https://user-images.githubusercontent.com/4208480/144266668-02c41b3b-5283-4c64-8b33-77a0331b33b5.png)

**Comment by igorkamyshev** [comment]
Yeah, it looks like a bug. I'll check it as soon as possible.

**Comment by Kelin2025** [comment]
The same with `.watch`, by the way:
```tsx
sample($nameInput, nameInputChanged).watch(console.log)
```
^ gives the same error

**Comment by igorkamyshev** [comment]
Thanks 🙏

**Comment by igorkamyshev** [comment]
Fixed in 0.4.2
